### PR TITLE
Implement __hash__ for GraphQLError

### LINKED ISSUE #34: GraphQLError is unhashable
It seems that the `logging` library in python assumes that exceptions are hashable, in order to be logged.  It'd be great if we could treat GraphQL errors the same way as other builtin exceptions.

See also, a similar issue in the `schematics` project:

https://github.com/schematics/schematics/issues/452

### PR DESCRIPTION
**Pull request by andrew-humu**
Fixes #34 

This seemed like the most minimal implementation that might be safe, but it's definitely possible there's some context I'm missing.  I also claim no mastery of python!

### CODE REVIEW

**Comment by Cito** [comment]
Thank your for providing this PR. Hashing only `original_error` assumes that one original error cannot cause two different GraphQLErrors, which is generally true, and that it is always set, which is not always the case. It would be better to derive the hash from all slotted attributes.

But I think the best and most performant solution is to simply use the default hash method for exceptions, based on the id of the instance. GraphQLError lost that default hash capability when it got a custom `__eq__` method, but the default hash method can be restored by setting `__hash__ = Exception.__hash__`. Can you check if that works for you and change the PR accordingly?

**Comment by andrew-humu** [comment]
Your reasoning makes a lot of sense to me.  I pushed a commit — is this what you had in mind?

File: tests/error/test_graphql_error.py [hunk 1]
```diff
@@ -128,9 +128,10 @@ def default_error_formatter_includes_extension_fields():
...
 
     def is_hashable():
-        try:
-            raise RuntimeError("original")
-        except RuntimeError as e:
-            original = e
-        e = GraphQLError("msg", original_error=original)
-        assert hash(e) == hash(original)
+        e = GraphQLError("msg", extensions={"foo": "bar"})
+        assert hash(e) is not Nonw
```
Comment (jplock) [inline tests/error/test_graphql_error.py]: `None`?

Comment (Cito) [reply tests/error/test_graphql_error.py]: Python would barf when returning None anyway. Maybe just a one-liner: `hash(GraphQLError("msg"))`

File: tests/error/test_graphql_error.py [hunk 2]
```diff
@@ -128,9 +128,10 @@ def default_error_formatter_includes_extension_fields():
...
+
+    def hashes_are_unique():
```
Comment (Cito) [inline tests/error/test_graphql_error.py]: Maybe `hashes_are_unique_per_instance` to make this clearer.

File: tests/error/test_graphql_error.py [hunk 3]
```diff
@@ -128,9 +128,10 @@ def default_error_formatter_includes_extension_fields():
...
+        e1 = GraphQLError("msg", extensions={"foo": "bar"})
```
Comment (Cito) [inline tests/error/test_graphql_error.py]: Maybe remove the extensions arg as it plays no particular role here.

File: graphql/error/graphql_error.py [hunk 4]
```diff
@@ -166,7 +166,7 @@ def __eq__(self, other):
         )
 
     def __hash__(self):
-        return hash(self.original_error)
+        return super(GraphQLError, self).__hash__()
```
Comment (Cito) [inline graphql/error/graphql_error.py]: That should be just `super().__hash__()` in Python 3. Or even remove the `def` and simply set  `__hash__ = Exception.__hash__` here.

**Pull request merged**